agent/consul/internal_endpoint.go

```diff
@@ -433,6 +437,208 @@ func (m *Internal) EventFire(args *structs.EventFireRequest,
 	return errs
 }
 
+// TODO: docs
+func (m *Internal) ReachabilityProbe(
+	args *structs.ReachabilityRequest,
+	reply *structs.ReachabilityResponses) error {
+	if done, err := m.srv.ForwardRPC("Internal.ReachabilityProbe", args, args, reply); done {
+		return err
+	}
+
+	// Check ACLs
+	rule, err := m.srv.ResolveToken(args.Token)
+	if err != nil {
+		return err
+	}
+	if rule != nil && rule.OperatorRead(nil) != acl.Allow {
```

Comment:
Short of inventing a new ACL resource, this seemed appropriate. Doing a filtered `node` style (like catalog API endpoints) seemed wrong as it would omit nodes from consideration during the diagnostic.

Reply:
`operator:read` seems perfect.